background.js

```diff
@@ -133,79 +90,66 @@ function modifyReferer(e)
 
 
 /*
- * This function is called when the inject-referer.js content script
- * asks for the value to set as document.referrer.
+ * Listener function for storage changes: Forward any changes to the
+ * global variables.
  */
-function handleContentMsg(request, sender, sendResponse) {
-	const conf = findHostConf(request.target, request.referrer);
-	var referrer = null;
-	switch (conf.action)
-	{
-		case "prune":
-			referrer = new URL(request.referrer).origin + "/";
-			break;
-		case "target":
-			referrer = new URL(request.target).origin + "/";
-			break;
-		case "replace":
-			referrer = conf.referer;
-			break;
-		case "remove":
-			referrer = '';
-			break;
-		default:
-			/* "keep" */
-			referrer = request.referrer;
+async function refreshConfig(change, area)
+{
+	let dirty = false;
+	if (area === "sync") {
```

Comment:
`"sync"` is a magic constant here, which is… not so good.

Reply:
The old code used "sync" here.